opcua/adapter.go

```diff
@@ -102,37 +103,62 @@ func (as *adapterService) RemoveChannel(ctx context.Context, chanID string) erro
 	return as.channelsRM.Remove(ctx, chanID)
 }
 
-func (as *adapterService) ConnectThing(ctx context.Context, chanID, thingID string) error {
+func (as *adapterService) ConnectThing(ctx context.Context, chanID string, thingIDs []string) error {
 	serverURI, err := as.channelsRM.Get(ctx, chanID)
 	if err != nil {
 		return err
 	}
 
-	nodeID, err := as.thingsRM.Get(ctx, thingID)
-	if err != nil {
-		return err
-	}
+	for _, t := range thingIDs {
+		nodeID, err := as.thingsRM.Get(ctx, t)
+		if err != nil {
+			return err
+		}
 
-	as.cfg.NodeID = nodeID
-	as.cfg.ServerURI = serverURI
+		as.cfg.NodeID = nodeID
+		as.cfg.ServerURI = serverURI
 
-	c := fmt.Sprintf("%s:%s", chanID, thingID)
-	if err := as.connectRM.Save(ctx, c, c); err != nil {
-		return err
-	}
+		c := fmt.Sprintf("%s:%s", chanID, t)
+		if err := as.connectRM.Save(ctx, c, c); err != nil {
+			return err
+		}
 
-	go func() {
-		if err := as.subscriber.Subscribe(ctx, as.cfg); err != nil {
-			as.logger.Warn(fmt.Sprintf("subscription failed: %s", err))
+		go func() {
+			if err := as.subscriber.Subscribe(ctx, as.cfg); err != nil {
+				as.logger.Warn(fmt.Sprintf("subscription failed: %s", err))
+			}
+		}()
+
+		// Store subscription details
+		if err := db.Save(serverURI, nodeID); err != nil {
+			return err
 		}
-	}()
+	}
 
-	// Store subscription details
-	return db.Save(serverURI, nodeID)
+	return nil
 }
 
-func (as *adapterService) Browse(ctx context.Context, serverURI, namespace, identifier string) ([]BrowsedNode, error) {
-	nodeID := fmt.Sprintf("%s;%s", namespace, identifier)
+func (as *adapterService) Browse(ctx context.Context, serverURI, namespace, identifier, identifierType string) ([]BrowsedNode, error) {
+	var nodeID string
+
+	switch identifierType {
+	case "string":
+		nodeID = fmt.Sprintf("ns=%s;s=%s", namespace, identifier)
+	case "numeric":
+		numericIdentifier, err := strconv.Atoi(identifier) // Convert identifier to int
+		if err != nil {
+			nodeID = fmt.Sprintf("ns=%s;s=%s", namespace, identifier)
+			as.logger.Warn(fmt.Sprintf("failed to parse numeric nodeID format: %s, defaulting to string", err))
+			break
+		}
```

Comment:
Please pass arguments to logger instead of fmt.Sprintf(), Arguements gives us json output.
For instance
```suggestion
		if err != nil {
			args := []any{
				slog.String("namespace", namespace),
				slog.String("identifier", identifier),
				slog.String("error", err.Error()),
			}
			as.logger.Warn("failed to parse numeric identifier", args...)
			break
		}
```